### ISSUE TITLE
Support TLS 1.2

### ISSUE THREAD

**Issue by duncanrand** [ic-hep/pdm#450]
An error occurred during a connection. Peer using unsupported version of security protocol.

Error code: SSL_ERROR_UNSUPPORTED_VERSION

**Comment by sfayer** [comment]
This is a property of the pyOpenSSL build/version... On the (stock) CentOS7 version, this only supports up to TLSv1.1. The proper fix for this would be to move to python3 and deploy on CentOS8 instead (where it would then have support up to TLSv1.3). gfal is available on CentOS8 now and Alex has written a python3 patch, so this may well be possible, although no-one has tried it yet. Perhaps you could deploy on CentOS8 instead and we'll fix up any issues that you find as you go along?

**Comment by duncanrand** [comment]
On CentOS8 I get

Error: Unable to find a match: python-virtualenv m2crypto gfal2-python

**Comment by alexanderrichards** [comment]
* you don't need python-virtualenv with python3 as it is built in. 
* m2crypto may be now python3-m2crypto
* gfal2-python I don't know.... try listing python3-g*

**Comment by sfayer** [comment]
Indeed: `python3-gfal2` & `python3-m2crypto`

**Comment by duncanrand** [comment]
Problem: conflicting requests
   - nothing provides libboost_python3.so.1.66.0()(64bit) needed by 
python3-gfal2-1.9.5-5.el8.x86_64

Here it suggests PowerTools:

https://bugzilla.redhat.com/show_bug.cgi?id=1765854

but

$ sudo dnf config-manager --set-enabled PowerTools
Error: No matching repo to modify: PowerTools.

**Comment by sfayer** [comment]
Yes, it is in powertools although exactly what the repo is called on the cloud image, I don't know. Have a poke about in `/etc/yum.repos.d` and see if you can find it in there.

**Comment by duncanrand** [comment]
sudo dnf config-manager --set-enabled powertools

works.
